FreeRTOS_RA.c

```diff
@@ -88,60 +119,76 @@
         size_t uxNeededSize;
         MACAddress_t xMultiCastMacAddress;
         NetworkBufferDescriptor_t * pxDescriptor = pxNetworkBuffer;
+        IPv6_Address_t xSourceAddress;
+        BaseType_t xHasLocal;
 
         configASSERT( pxEndPoint != NULL );
-        uxNeededSize = ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER + sizeof( ICMPRouterSolicitation_IPv6_t );
 
-        if( pxDescriptor->xDataLength < uxNeededSize )
+        xHasLocal = xGetlinklocaladdress( pxEndPoint->pxNetworkInterface, &( xSourceAddress ) );
+
+        if( xHasLocal == pdFAIL )
         {
-            pxDescriptor = pxDuplicateNetworkBufferWithDescriptor( pxDescriptor, uxNeededSize );
+            FreeRTOS_printf( ( "RA: can not find a Link-local address\n" ) );
```

Comment:
Shall we use local IP-address when link-local address is not found?

Reply:
If it does not have a local-link address, I propose to use the "any address", which is all zero's: "`::`".
Just tested: my router also responds to the "any address".